core/src/main/scala/org/genivi/sota/core/VehicleUpdatesResource.scala

```diff
@@ -166,16 +166,22 @@ class VehicleUpdatesResource(db : Database, resolverClient: ExternalResolverClie
 
   val route = {
     (pathPrefix("api" / "v1" / "vehicle_updates") & extractVin) { vin =>
-      (put & path("installed")) { updateInstalledPackages(vin) } ~
-      (put & path("order")) { setInstallOrder(vin) } ~
-      (put & extractUuid & path("cancelupdate") ) { uuid => cancelUpdate(vin, uuid) } ~
-      (post & namespaceExtractor & pathEnd) { ns => queueVehicleUpdate(ns, vin) } ~
-      (get & pathEnd) { logVehicleSeen(vin) { pendingPackages(vin) } } ~
-      (get & path("queued")) { pendingPackages(vin) } ~
-      (get & extractUuid & path("download")) { uuid => downloadPackage(vin, uuid) } ~
-      (get & path("operationresults")) { results(vin) } ~
-      (post & extractUuid) { reportInstall } ~
-      (post & path("sync")) { sync(vin) }
+      get {
+        pathEnd { logVehicleSeen(vin) { pendingPackages(vin) } } ~
+        path("queued") { pendingPackages(vin) } ~
+        path("operationresults") { results(vin) } ~
+        (extractUuid & path("download")) { uuid => downloadPackage(vin, uuid) }
+      } ~
+      put {
+        path("installed") { updateInstalledPackages(vin) } ~
+        path("order") { setInstallOrder(vin) } ~
+        (extractUuid & path("cancelupdate") ) { uuid => cancelUpdate(vin, uuid) }
+      } ~
+      post {
+        path("sync") { sync(vin) } ~
+        (namespaceExtractor & pathEnd) { ns => queueVehicleUpdate(ns, vin) } ~
+        (extractUuid & pathEnd) { reportInstall }
```

Comment:
This arrangement keeps route-matching simple for the Routing DSL.
The previous arrangement failed in non-obvious ways.
Probably another arrangement would be more readable, but given past DSL behavior that could also be more error-prone.